Update installation guide for Sublime (#1597)
scalameta/metals: docs/editors/sublime.md

title: Sublime Text
---

> Metals works with Sublime Text 3 thanks to

Reviewer: ```suggestion
Metals works with Sublime Text 3 thanks to the
```

[sublimelsp/LSP](https://github.com/sublimelsp/LSP) and [scalameta/metals-sublime](https://github.com/scalameta/metals-sublime) plugins.

![Sublime Text demo](https://i.imgur.com/vJKP0T3.gif)

```scala mdoc:requirements

```

## Installing the plugins

Install the following packages:

- [sublimelsp/LSP](https://github.com/sublimelsp/LSP): Language Server Protocol support for Sublime Text.  
`Command Palette (Cmd + Shift + P) > Install package > LSP`

> - [scalameta/metals-sublime](https://github.com/scalameta/metals-sublime): For automatic installation of metals and custom endpoints. 

Reviewer: ```suggestion
- [scalameta/metals-sublime](https://github.com/scalameta/metals-sublime): For automatic installation of metals and custom commands. 
```
I'm not 100% sure what endpoints mean in this context. Is commands clearer? What do you think?

Author: It means commands like build-import, doctor-run, you are right it might be clearer

]
```

## Importing a build

> Open Sublime in the base directory of an sbt build and it will then prompt you to import the build. Click "Import build" to start the installation step.

Reviewer: ```suggestion
Open Sublime in the base directory of your Scala project and it will then prompt you to import the build as long as you're using one of the [supported build tools](https://scalameta.org/metals/docs/build-tools/overview.html). Click "Import build" to start the installation step.
```
What do you think about not being so specific to sbt? I just looked and we phrased it this way in the vscode docs as well, but I sort of feel like this is more accurate. What do you think?

> This starts the Metal language server but no functionality will work because the

Reviewer: ```suggestion
This starts the Metal language server but no functionality will work yet because the
```

build has not been imported. 

This step can take a long time, especially the first time you run it in a new
workspace. The exact time depends on the complexity of the build and if the library dependencies are cached or need to be downloaded. For example, this step can take anywhere from 10 seconds in small cached builds up to 10-15 minutes in large un-cached builds.

For more detailed information about what is happening behind the scenes during

> `sbt bloopInstall` run `lsp toggle server panel` in command palette you can optionally add key binding for this command.

Reviewer: ```suggestion
`sbt bloopInstall` run `lsp toggle server panel` in the command palette. You can optionally add key binding for this command.
```

## Goto symbol in workspace

> You can search for symbols in your dependency source using the command palette

Reviewer: ```suggestion
You can search for symbols in your dependency source using the command palette.
```

You can search for symbols in your dependency source using the command palette

![workspace symbols](https://i.imgur.com/8X0XNi2.gif)

## Manually trigger build import

> You can configure a custom command "Metals: Import Build" to manually trigger

Reviewer: I'm assuming that this is no longer needed?

Author: Yes it is part of the package commands [here](https://github.com/scalameta/metals-sublime/blob/master/metals-sublime.sublime-commands)

Reviewer: Cool, that's what I figured. 👍
